src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -826,17 +825,29 @@ private void ProcessSettingsFrame(FrameHeader frameHeader, bool initialFrame = f
                             // We don't actually store this value; we always send frames of the minimum size (16K).
                             break;
 
+                        case SettingId.EnableConnect:
+                            if (settingValue == 1)
+                            {
+                                IsConnectEnabled = true;
+                            }
+                            break;
+
                         default:
                             // All others are ignored because we don't care about them.
                             // Note, per RFC, unknown settings IDs should be ignored.
                             break;
                     }
                 }
 
-                if (initialFrame && !maxConcurrentStreamsReceived)
+                if (initialFrame)
                 {
-                    // Set to 'infinite' because MaxConcurrentStreams was not set on the initial SETTINGS frame.
-                    ChangeMaxConcurrentStreams(int.MaxValue);
+                    if (!maxConcurrentStreamsReceived)
+                    {
+                        // Set to 'infinite' because MaxConcurrentStreams was not set on the initial SETTINGS frame.
+                        ChangeMaxConcurrentStreams(int.MaxValue);
+                    }
+
+                    InitialSettingsReceived.TrySetResult(true);
```

Comment:
```suggestion
                    if (_initialSettingsReceived is null)
                    {
                        Interlocked.CompareExchange(_initialSettingsReceived, s_settingsReceivedSingleton, null);
                    }
                    InitialSettingsReceived.TrySetResult(true);
```

Reply:
Should I put `InitialSettingsReceived.TrySetResult(true)` under `else`?

Reply:
Nope. You want to call it in case the CompareExchange loses the race. It could avoid the overhead if CompareExchange wins the race, but I don't think it's worth the complication.